File: 백준/Silver/10845. 큐/큐.py
from collections import deque
import sys
input = sys.stdin.readline

qq = deque([])
n = int(input())

for i in range(n):
    command = input().split()

    if command[0] == 'push':
        qq.append(int(command[1]))
    elif command[0] == 'pop':
        if len(qq) == 0:
            print(-1)
        else:
            print(qq.popleft())
    elif command[0] == 'size':
        print(len(qq))
    elif command[0] == 'empty':
        if len(qq) == 0:
            print(1)
        else:
            print(0)
    elif command[0] == 'front':
        if len(qq) == 0:
            print(-1)
        else:
            print(qq[0])
    elif command[0] == 'back':
        if len(qq) == 0:
            print(-1)
        else:
            print(qq[-1])

File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        Deque<Integer> q = new ArrayDeque<>();
        for(int i=0; i<n; i++) {
            String[] inputs = br.readLine().split(" ");
            if(inputs[0].equals("push")) {
                q.offer(Integer.valueOf(inputs[1]));
            } else if (inputs[0].equals("pop")) {
                if(q.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(q.peek());
                    q.poll();
                }
            } else if (inputs[0].equals("size")) {
                System.out.println(q.size());
            } else if (inputs[0].equals("empty")) {
                if(q.isEmpty()) {
                    System.out.println(1);
                } else {
                    System.out.println(0);
                }
            } else if (inputs[0].equals("front")) {
                if(q.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(q.peekFirst());
                }
            } else if (inputs[0].equals("back")) {
                if(q.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(q.peekLast());
                }
            }
        }
    }
}